File: codegen/results/v002/m3-max-48gb/ollama/qwen2.5-coder_7b/20260411_224426/typescript/008_prime_numbers/solution.ts
// Function to check if a number is prime
function isPrime(num: number): boolean {
    if (num <= 1) return false;
    if (num <= 3) return true;

    if (num % 2 === 0 || num % 3 === 0) return false;

    for (let i = 5; i * i <= num; i += 6) {
        if (num % i === 0 || num % (i + 2) === 0) return false;
    }
    return true;
}

// Main function to find and print prime numbers up to N
function findAndPrintPrimes(N: number): void {
    for (let i = 2; i <= N; i++) {
        if (isPrime(i)) {
            console.log(i);
        }
    }
}

// Entry point of the program
if (process.argv.length < 3) {
    console.error("Usage: tsx script.ts <N>");
    process.exit(1);
}

const N = parseInt(process.argv[2], 10);

if (isNaN(N)) {
    console.error("Invalid input. Please provide a valid integer.");
    process.exit(1);
}

findAndPrintPrimes(N);

File: codegen/results/v002/m3-max-48gb/ollama/qwen2.5-coder_7b/20260411_224426/go/008_prime_numbers/solution.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

// Helper function to check if a number is prime
func isPrime(n int) bool {
	if n <= 1 {
		return false
	}
	for i := 2; i*i <= n; i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func main() {
	// Check if the command line argument is provided and convert it to an integer
	if len(os.Args) != 2 {
		fmt.Println("Usage: go run program.go <integer>")
		os.Exit(1)
	}

	nStr := os.Args[1]
	n, err := strconv.Atoi(nStr)
	if err != nil {
		fmt.Println("Please provide a valid integer.")
		os.Exit(1)
	}

	// Find and print all prime numbers up to and including N
	for i := 2; i <= n; i++ {
		if isPrime(i) {
			fmt.Println(i)
		}
	}
}